pkg/util/mount/mount_windows.go

```diff
@@ -211,18 +211,23 @@ func (mounter *Mounter) GetFileType(pathname string) (FileType, error) {
 		return pathType, err
 	}
 
-	mode := info.Sys().(*syscall.Win32FileAttributeData).FileAttributes
-	switch mode & syscall.S_IFMT {
-	case syscall.S_IFSOCK:
-		return FileTypeSocket, nil
-	case syscall.S_IFBLK:
-		return FileTypeBlockDev, nil
-	case syscall.S_IFCHR:
-		return FileTypeCharDev, nil
-	case syscall.S_IFDIR:
+	// checks whether the mode is the target mode
+	isSpecificMode := func(mode, targetMode os.FileMode) bool {
+		return mode&targetMode == targetMode
+	}
+
+	mode := info.Mode()
```

Comment:
Can this be done for linux too?

Reply:
Yes. Works for linux as well. Will create a common function for use.